### ISSUE TITLE
db-custom-select: focus ring doesn't work correctly

### ISSUE THREAD

**Issue by dnepro** [db-ux-design-system/core-web#4916]
### Which generators are affected?

- [x] All
- [ ] HTML
- [ ] React
- [ ] Angular
- [ ] Vue
- [ ] Web components
- [ ] Power Apps

### Reproduction case

Open e.g. https://design-system.deutschebahn.com/core-web/version/v3.0.8/components/data-input/custom-select/overview and interact with one of the custom select elements with a pointer device like a mouse instead of a keyboard. A keyboard focus ring will appear unexpectedly.

### Expected Behaviour

Focus ring is initially on first item (even though there was no keyboard or other interaction - shouldn't be focused)
when navigation the options the focus ring doesn't follow

### Screenshots

https://www.figma.com/board/Tcc3RmH42UnnuLLlanG3oE/Help--Hints---Hugs?node-id=4008-13182&t=RCsF3O0oGnMCimKl-4

### Browser version

None

### Which DB business unit do you work for?

DB InfraGO AG

### What project are you working on?

prep2plan
